docs/tutorials/python/table.md

```diff
@@ -1,2 +1,217 @@
+)
+for _, row_info in results.iterrows():
+    print(row_info)
+```
+
+
+## 3. Changing Data
+
+Once the schema is settled, changes come in two flavors: appending new rows and updating existing ones.
+
+### Appending new rows
+
+Appending new rows is fairly straightforward. To continue the previous example, we might add some new genes from another file.
+Lets take this new data in `more_genes.csv`
+
+```csv
+Name,Chromosome,Start,End,Strand,TranscriptionFactor
+Qux1,4,201001,202001,+,False
+Qux2,4,203001,204001,+,False
+```
+
+* Using a CSV
+
+    ```python
+    table = Table(
+        name="My Favorite Genes",
+        parent_id=project.id,
+    ).get()
+    table = table.store_rows(values="/path/to/more_genes.csv")
+    ```
+
+* Using Pandas
+
+    ```python
+    new_rows_df = pd.DataFrame({
+        "Name": ["Qux3", "Qux4"],
+        "Chromosome": ["4", "4"],
+        "Start": [201001, 203001],
+        "End": [202001, 204001],
+        "Strand": ["+", "+"],
+        "TranscriptionFactor": [False, False]
+    })
+    table.store_rows(values=new_rows_df)
+    ```
+
+### Updating existing rows
+
+* Updating the existing table - query the existing table and update the data. Minimizing changesets to contain only rows that actually change will make processing faster.
+
+    ```python
+    results_df = query(f"select * from {table.id} where Chromosome='1'")
+    results_df['Name'] = ['rzing', 'zing1', 'zing2', 'zing3']
+    table.store_rows(values=results_df)
+    ```
+
+* Upserting rows (update or insert) - If your data has a primary key, you can use the upsert functionality to update existing rows or insert new rows based on this primary key.  This way you won't have to query for the existing data to update your table. This function does not do deletions.
+
+    ```python
+    to_upsert_df = pd.DataFrame({
+        "Name": ["Qux3", "Qux5"],
+        "Chromosome": ["4", "4"],
+        "Start": [201001, 203001],
+        "End": [202001, 204001],
+        "Strand": ["-", "+"],
+        "TranscriptionFactor": [True, False]
+    })
+    # Qux3 will be updated, Qux5 will be inserted
+    table.upsert_rows(values=to_upsert_df,  primary_keys=['Name'])
+    ```
+
+## 4. Changing Table Structure
+
+* Adding columns
+
+    ```python
+    table.add_column(
+        Column(name="Expression", column_type="STRING")
+    )
+    table.store()
+    ```
+
+* Renaming or modifying a column:
+
+    ```python
+    table.columns['Expression'].name = 'Expression2'
+    table.columns['Expression'].column_type = 'INTEGER'
+    table.store()
+    ```
+
+* Removing a column
+
+    ```python
+    table.delete_column(name="Expression2")
+    table.store()
+    ```
+
+## 5. Deleting Table rows & Tables
+
+* Deleting specific rows - Query for the rows you want to delete and call syn.delete on the results
+
+    ```python
+    table.delete_rows(query=f"SELECT * FROM {table.id} WHERE Strand = '+'")
```

Comment:
I think we may want to work on being able to delete a file based on a filtered dataframe.  Take for example:

1. do the table query
2. do complex filtering (not with SQL) to find cols to delete
3. table.delete_rows(values=df)

we can gage interest

Reply:
That should be really easy to implement, basically instead of running this line:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/15a2b61f237ffd1a993c0435a4492748b6c18260/synapseclient/models/mixins/table_components.py#L4114

We just take the DataFrame the user passes into the method call instead. That's it. We would probably want to do a small conditional check to verify that the DataFrame has the required columns.